mysql-test/extra/binlog_tests/binlog_drop_tbl_with_comment.test

```diff
@@ -0,0 +1,56 @@
+#
+# PS-7526: Unexpected quoting and dropping of comments in DROP TABLE commands
+#
+
+RESET MASTER;
+CREATE DATABASE test7526;
+
+USE test7526;
+
+/* comment1 */ CREATE TABLE t (a VARCHAR(100));
+/* comment1 */ DROP TABLE t;
+
+CREATE TABLE t2 (a VARCHAR(100));
+DROP TABLE t2 /* comment at end */;
+
+SET binlog_ddl_skip_rewrite = ON;
+
+/* comment1 */ CREATE TABLE t (a VARCHAR(100));
+/* comment1 */ DROP TABLE t;
+
+CREATE TABLE t2 (a VARCHAR(100));
+DROP TABLE t2 /* comment at end */;
+
+SET binlog_ddl_skip_rewrite = OFF;
+
+/* comment1 */ CREATE TABLE t (a VARCHAR(100));
+/* comment1 */ DROP TABLE t;
```

Comment:
While analyzing test results, I just realized, we didn't add a test for partitioned table. and also drop multi tables of partition type within a version comment marker

Can you please add these?

Test results look good now